### PR TITLE
Backports 0.15 pr19

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
shouldn't conflict at all with multiwallet pr

### CODE REVIEW

File: test/functional/test_framework/util.py [hunk 1]
```diff
@@ -205,21 +205,40 @@ def initialize_datadir(dirname, n):
...
-
-def rpc_url(i, rpchost=None):
-    rpc_u, rpc_p = rpc_auth_pair(i)
+def get_datadir_path(dirname, n):
+    return os.path.join(dirname, "node"+str(n))
+
+def get_auth_cookie(datadir, n):
+    user = None
+    password = None
+    if os.path.isfile(os.path.join(datadir, "bitcoin.conf")):
```
Comment (UdjinM6) [inline test/functional/test_framework/util.py]:
```suggestion
    if os.path.isfile(os.path.join(datadir, "dash.conf")):
```

File: test/functional/test_framework/util.py [hunk 2]
```diff
@@ -205,21 +205,40 @@ def initialize_datadir(dirname, n):
...
+        with open(os.path.join(datadir, "bitcoin.conf"), 'r') as f:
```
Comment (UdjinM6) [inline test/functional/test_framework/util.py]:
```suggestion
        with open(os.path.join(datadir, "dash.conf"), 'r') as f:
```

File: src/util.h [hunk 3]
```diff
@@ -241,62 +241,63 @@ class ArgsManager
...
+     * @param strArg Argument to set (e.g. "-foo")
+     * @param fValue Value (e.g. false)
+     * @return true if argument gets set, false if it already had a value
+     */
+    bool SoftSetBoolArg(const std::string& strArg, bool fValue);
+
+    // Forces an arg setting. Called by SoftSetArg() if the arg hasn't already
+    // been set. Also called directly in testing.
+    void ForceSetArg(const std::string& strArg, const std::string& strValue);
 void ForceSetMultiArgs(const std::string& strArg, const std::vector<std::string>& values);
```
Comment (UdjinM6) [inline src/util.h]:
```suggestion
    void ForceSetMultiArgs(const std::string& strArg, const std::vector<std::string>& values);
```

File: src/util.h [hunk 4]
```diff
@@ -241,62 +241,63 @@ class ArgsManager
...
 void ForceRemoveArg(const std::string& strArg);
```
Comment (UdjinM6) [inline src/util.h]:
```suggestion
    void ForceRemoveArg(const std::string& strArg);
```

**Comment by UdjinM6** [comment]
Still need 10759 (https://github.com/dashpay/dash/pull/3023#discussion_r301983698) to fix `multi_rpc.py` test https://travis-ci.org/dashpay/dash/jobs/556841057#L2719

**Comment by PastaPastaPasta** [comment]
was already in working branch, just had to cherry-pick in

**Pull request merged**